### ISSUE TITLE
Having multiple mounts makes everything rather slow

### ISSUE THREAD

**Issue by netzvieh** [gopasspw/gopass#1520]
### Summary
Most gopass actions get very slow when using multiple mounts:

```
time gopass ls --flat >/dev/null

real    0m1,186s
user    0m1,148s
sys     0m0,067s
```

This is 13 secrets in 7 mounts. It doesn't increase with the number of secrets (having 200+ secrets results in roughly the same time needed). But does increase with number of mounts.

Adding 4 empty mounts results in:

```
time gopass ls --flat >/dev/null

real    0m1,817s
user    0m1,755s
sys     0m0,106s
```

### Steps To Reproduce
Create a store with multiple mounts:

```
gopass init
for i in {1..10}; do
  gopass init --path /tmp/foo$i --store foo$i
done
gopass ls --flat
```

### Expected behavior
Don't add excessive amounts of time per mount.

### Environment
- OS: OpenSUSE Tumbleweed
- OS version: Linux 5.8.0-1-default #1 SMP Tue Aug 4 07:30:59 UTC 2020 (9bc0044) x86_64 x86_64 x86_64 GNU/Linux
- gopass Version: gopass 1.9.2+e2d1549f452a0df1fc52e42e7d0f654334d7144e (e2d1549f452a0df1fc52e42e7d0f654334d7144e) go1.14.2 linux amd64
- Installation method: Github Release

### Additional context
I'll try to add a debug output as a file.

**Comment by netzvieh** [comment]
[gopass_mounts_slow_debug.txt](https://github.com/gopasspw/gopass/files/5073806/gopass_mounts_slow_debug.txt)

**Comment by dominikschulz** [comment]
Interesting. Thanks for providing a detailed bug report.
I've never noticed this behaviour.
But since having multiple mounts is certainly one of our core features we should try to make it scale reasonably well.

So far we've never formally defined how many mounts we want to support, but we should document the expecations around that.

Also we might want to add timestamps to the debug logs.

**Comment by AnomalRoil** [comment]
We might want to document that using 4096 bit keys makes everything slower as well. :)
I recently made a keysize change and really noticed the difference.

**Comment by dominikschulz** [comment]
But that's GPG. I won't take any responsibility for that ;)

But we should definitely document that.

**Comment by dominikschulz** [comment]
So, I did look into this a bit.
I couldn't really reproduce anything that I'd see as a problem (wrt. performance), i.e. for me it works very well even with dozens of stores, hundreds of entries each.

However after adding higher granularity timestamps into the debug logs I found something that might degrade performance on some systems.

When we initialize a GPG mount we always detect the GPG binary from scratch (this was introduced once to work around some GPG quirks). I can only imagine that on some systems with a lot of binaries and possible slow IO this might degrade poorly. It's perfectly fine on my decade-old machine, but it might be bad on some.

So this is another GPG bug ... *sigh*.

But I have added *yet another annoying GPG workaround* to speed things up, improved our logging and tried to document our first core use case (I'm planning to document our other core use cases as well soon).
